### PR TITLE
DTSPO-8837 - Azurerm v3 provider backend app gw

### PR DESCRIPTION
**Pull request by endakelly**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-8837

### Change description ###
Use azurerm v3 provider for backend app gw
Once azure-platform-terraform repo has also been updated, the branches on the modules will be merged to master and the references to them in this repo will be reverted to master.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: environments/dev/backend_lb_config.yaml [hunk 1]
```diff
@@ -11,25 +11,33 @@ gateways:
     # Toffee application
       - product: toffee
         component: recipe-backend
+        priority: 10
```
Comment (timja) [inline environments/dev/backend_lb_config.yaml:14]: what's this?

Comment (endakelly) [reply environments/dev/backend_lb_config.yaml:14]: This has to be set in the latest azurerm api for some reason. In previous versions this is set automatically but now it has to be done manually. Haven't yet figured out how to generate the figure dynamically so it doesn't need to be coded in the vars
https://github.com/hashicorp/terraform-provider-azurerm/issues/17079

Comment (timja) [reply environments/dev/backend_lb_config.yaml:14]: is there some way to get an index and then add 10?

Comment (endakelly) [reply environments/dev/backend_lb_config.yaml:14]: Matt sent me a code snippet that I'm going to look to adapt for this

**Review by timja — APPROVED** [review]

**Pull request merged**